mlflow/langchain/__init__.py

```
@@ -393,14 +575,62 @@ def _save_model(model, path):
     return model_data_kwargs
 
 
-def _load_model(path, agent_path=None, tools_path=None, agent_primitive_path=None):
-    model = None
-    if agent_path is None and tools_path is None:
-        from langchain.chains.loading import load_chain
+def _load_from_pickle(loader_fn_path, persist_dir):
+    if loader_fn_path is None:
+        raise mlflow.MlflowException.invalid_parameter_value(
+            "Missing file for loader_fn which is required to build the model."
+        )
```

I think we can throw this error in `_load_model`, not here.